cse-lib/network-processing/src/main/java/com/farao_community/farao/cse/network_processing/busbar_change/NetworkHelper.java

```diff
@@ -99,28 +99,45 @@ private static boolean isBranchConnected(String branchId, String nodeId, Network
 
         VoltageLevel voltageLevel = ((Bus) network.getIdentifiable(node1)).getVoltageLevel();
 
-        // Create fictitious bus
-        String busId = generateFictitiousBusId(voltageLevel, network);
-        Bus fictitiousBus = networkModifier.createBus(busId, voltageLevel.getId());
+        boolean moveSide1 = bus1.equals(node1) || bus1.equals(node2);
+        boolean moveSide2 = bus2.equals(node1) || bus2.equals(node2);
 
-        // Move one branch end to the fictitious bus
         boolean branchIsOnNode1 = true; // check if branch is initially on node1
-        if (bus1.equals(node1) || bus1.equals(node2)) {
-            networkModifier.moveBranch(branch, Branch.Side.ONE, fictitiousBus);
+        if (moveSide1) {
             branchIsOnNode1 = bus1.equals(node1);
-        } else if (bus2.equals(node1) || bus2.equals(node2)) {
-            networkModifier.moveBranch(branch, Branch.Side.TWO, fictitiousBus);
+        } else if (moveSide2) {
             branchIsOnNode1 = bus2.equals(node1);
         }
-        // else: should not happen, a check was done before
+
+        String fictitiousBusId;
+        Bus fictitiousBus;
```

Comment:
You can remove that and use only fictitiousBusId below this block

Reply:
true